src/SignUp.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import axios from 'axios';
+import { useHistory } from 'react-router-dom';
+
+const url =
+  'http://ec2-13-209-5-166.ap-northeast-2.compute.amazonaws.com:8000/api/signup';
+
+export default function SignUp() {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [name, setName] = useState('');
+
+  let history = useHistory();
+
+  function handleChange(e, inputID) {
+    if (inputID === 0) setEmail(e.target.value);
+    else if (inputID === 1) setPassword(e.target.value);
+    else setName(e.target.value);
+  }
+  async function checkButtonHandle() {
+    if (email && password && name) {
+      try {
+        const response = await axios.post(url, {
+          email: email,
+          password: password,
+          name: name,
+        });
+        alert(response.data);
+        history.push('/signin');
+      } catch (error) {
+        if (error.response.status === 409) alert('이미 가입된 이메일입니다.');
+        else if (error.response.status === 400) alert('잘못된 요청입니다.');
+        else alert(error.response.data);
+      }
+    } else alert('빈칸 없이 입력해주세요');
+  }
+  return (
+    <Container>
+      <Title>회원가입</Title>
+      <InputText>
+        email
+        <InputForm value={email} onChange={(e) => handleChange(e, 0)} />
```

Comment:
```javascript
const ChangeHandler = (inputIndex) => (event) => {};
```
```suggestion
        <InputForm value={email} onChange={ChangeHandler(0)} />
```

이런 식으로 이벤트 핸들러를 직접 만드는 함수를 호출하면 인자 수와 핸들러의 복잡도를 줄일 수 있겠지?

`함수를 반환하는 함수`라는 개념은 다른 언어에서도 많이 쓰이니까(데코레이터 등) 잘 기억해둡시다~